Adds documentation on how to release varats and VaRA (#590)
se-passau/VaRA-Tool-Suite: docs/source/vara-ts/development.rst

It runs the command `benchbuild run -E JustCompile gzip` by default, which can be adapted to debug other projects and experiments by changing the arguments that are passed to `benchbuild`.
To step through the JustCompile experiment a breakpoint has to be set in just_compile.py.
`F9` can be used to set/unset a breakpoint at the current line.

> Releasing VaRA-TS and vara related tools/libraries

Reviewer: ```suggestion
Releasing VaRA-TS and VaRA related tools/libraries
```

..................................................

> In general, all VaRA related tools and libraries should have the same releases, which are compatible to each other, e.g., `vara-11.1.0` for vara-llvm-project needs to work with vara-tool-suite release `vara-11.1.0`.
> We try to prevent breaking changes to that the tools suite works with many different VaRA version, but sometimes breaking changes are necessary.

Reviewer: ```suggestion
In general, all VaRA related tools and libraries with the same release should be compatible with each other, e.g., `vara-11.1.0` for vara-llvm-project needs to work with vara-tool-suite release `vara-11.1.0`.
We try to prevent breaking changes so that the tool suite works with many different VaRA versions, but sometimes breaking changes between releases are necessary.
```

Prepare release tooling
^^^^^^^^^^^^^^^^^^^^^^^

> Before we can release anything, we need to install and prepare our release tools.

Reviewer: ```suggestion
Before we can create a release, we need to install and prepare our release tools.
```

Install:

.. code-block:: console

    pip install --user wheel tox

Configure pypi credentials

.. code-block:: none

    [distutils]
      index-servers =
        testpypi
        pypi

    [pypi]
      repository = https://upload.pypi.org/legacy/
      username = __token__
      password = $PYPI_PASSWORD

    [testpypi]
      repository = https://test.pypi.org/legacy/
      username = __token__
      password = $PYPI_PASSWORD

vara-tool-suite
^^^^^^^^^^^^^^^

> Ensure that all branches are on the right state.

Reviewer: ```suggestion
Ensure that all branches are in the correct state.
```

.. code-block:: console

    cd VaRA-Tool-Suite
    git pull vara
    git pull vara-dev

Integrate the changes from develop into the release branch `vara`.

.. code-block:: console

    git checkout vara
    git merge vara-dev

Build and upload release files with tox and tag the new release, this automatically builds both namespace packages `varats` and `varats-core`.

.. code-block:: console

    tox -e release
    git tag -s vara-X.Y.Z

    git push vara
    git push origin vara-X.Y.Z

Prepare the next version.

.. code-block:: console

    git checkout vara-dev
    # Update varats/setup.py and varats-core/setup.py to the next version
    # Update varats/setup.py to depend on the new core version

>     git commit --allow-empty -m "Bump version to $NEW_VERSION"

Reviewer: ```suggestion
    git commit -m "Bump version to $NEW_VERSION"
```

    git push vara-dev

vara-feature
^^^^^^^^^^^^

> Ensure that all branches are on the right state.

Reviewer: ```suggestion
Ensure that all branches are in the correct state.
```
